lib/url.js

```diff
@@ -1,6 +1,8 @@
 'use strict';
 
 const punycode = require('punycode');
+const icu = process.binding('config').hasIntl ? process.binding('icu') :
+                                                undefined;
```

Comment:
I'd change this so that it either imports icu or punycode but not both.  Saves a few kilobytes.

Also, importing the toASCII function directly should save a function call and property lookup, i.e.:
```js
function importPunycode() {
  try {
    return process.binding('icu');
  } catch (e) {
    return require('punycode');
  }
}

const { toASCII } = importPunycode();
```